File: ClipmanAndroid/app/src/main/java/me/onj/clipman/LinkClipboardText.kt
package me.onj.clipman

internal object LinkClipboardText {
    fun make(entry: ClipEntry, includeName: Boolean): String {
        val link = LinkPresentation.standaloneUrlText(entry.Text) ?: entry.Text.trim()
        val name = entry.Name.trim()
        return if (includeName && name.isNotEmpty()) "$name\n$link" else link
    }
}

File: ClipmanIOS/ClipmanIOS/Core/LinkClipboardText.swift
import Foundation

enum LinkClipboardText {
    static func make(entry: ClipEntry, url: URL, includeName: Bool) -> String {
        let link = url.absoluteString
        let name = entry.Name.trimmingCharacters(in: .whitespacesAndNewlines)
        guard includeName, !name.isEmpty else { return link }
        return "\(name)\n\(link)"
    }
}
